src/mscorlib/shared/System/IO/FileStream.Windows.cs

```diff
@@ -1735,5 +1735,77 @@ private void UnlockInternal(long position, long length)
                 throw Win32Marshal.GetExceptionForLastWin32Error();
             }
         }
+
+        private unsafe SafeFileHandle CreateFile2OpenHandle(FileMode mode, FileShare share, FileOptions options)
+        {
+            Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = GetSecAttrs(share);
+
+            int access =
+                ((_access & FileAccess.Read) == FileAccess.Read ? GENERIC_READ : 0) |
+                ((_access & FileAccess.Write) == FileAccess.Write ? GENERIC_WRITE : 0);
+
+            // Our Inheritable bit was stolen from Windows, but should be set in
+            // the security attributes class.  Don't leave this bit set.
+            share &= ~FileShare.Inheritable;
+
+            // Must use a valid Win32 constant here...
+            if (mode == FileMode.Append)
+                mode = FileMode.OpenOrCreate;
+
+            Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS parameters = new Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS();
+            parameters.dwSize = (uint)sizeof(Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS);
+            parameters.dwFileFlags = (uint)options;
+            parameters.lpSecurityAttributes = &secAttrs;
+
+            using (DisableMediaInsertionPrompt.Create())
+            {
+                if (AppDomain.IsAppXModel())
+                {
+                    return CreateFile2FromApp(_path, access, share, mode, ref parameters);
+                }
+                else
+                {
+                    return ValidateFileHandle(Interop.Kernel32.CreateFile2(
+                        lpFileName: _path,
+                        dwDesiredAccess: access,
+                        dwShareMode: share,
+                        dwCreationDisposition: mode,
+                        pCreateExParams: ref parameters));
+                }
+            }
+        }
+
+        // This needed dll won't be around for coreapp (only uap/uapaot) don't allow it to inline
```

Comment:
I do not understand what is this trying to achieve. I do not think it has any effect.